up-transport-vsomeip/src/transport_inner/transport_inner_engine.rs

```diff
@@ -0,0 +1,852 @@
+                        .get_pinned()
+                        .create_application(&app_name_cxx),
+                )
+            }
+        };
+        let Some(application_wrapper) = application_wrapper else {
+            return Err(UStatus::fail_with_code(
+                UCode::INTERNAL,
+                "Unable to create vsomeip application",
+            ));
+        };
+        application_wrapper.get_pinned().init();
+        (*application_wrapper).register_state_handler_fn_ptr_safe(available_state_handler_fn_ptr);
+
+        let app_name_start = app_name.to_string();
+        thread::spawn(move || {
+            let_cxx_string!(app_name_cxx = app_name_start.clone());
+            let runtime_wrapper = make_runtime_wrapper(vsomeip::runtime::get());
+
+            trace!(
+                "{}:{} - Within start_app, spawned dedicated thread to park app",
+                UP_CLIENT_VSOMEIP_TAG,
+                UP_CLIENT_VSOMEIP_FN_TAG_START_APP
+            );
+
+            let Some(application_wrapper) = make_application_wrapper(
+                runtime_wrapper.get_pinned().get_application(&app_name_cxx),
+            ) else {
+                return Err(UStatus::fail_with_code(
+                    UCode::INTERNAL,
+                    format!("No app found for app_name: {}", app_name_start),
+                ));
+            };
+
+            let client_id = application_wrapper.get_pinned().get_client();
+            trace!("start_app: after init'ing app we see its client_id: {client_id}");
+            application_wrapper.get_pinned().start();
+
+            Ok(())
+        });
+
+        trace!(
+            "{}:{} - Made it past creating app and starting it on dedicated thread",
+            UP_CLIENT_VSOMEIP_TAG,
+            UP_CLIENT_VSOMEIP_FN_TAG_START_APP
+        );
+
+        match receiver.recv_timeout(Duration::from_millis(50)) {
+            Err(err) => {
+                return Err(UStatus::fail_with_code(
+                    UCode::INTERNAL,
+                    format!("Timed out on waiting for application to start: {err:?}"),
+                ));
+            }
+            Ok(app_state) => {
+                info!("app_state: {app_state:?}");
+
+                application_wrapper.get_pinned().unregister_state_handler();
+                transport_storage
+                    .get_application_state_handler_registry()
+                    .free_state_handler_id(state_handler_id)?;
+            }
+        }
+
+        Ok(())
+    }
+
+    async fn event_loop(
+        ue_id: UeId,
+        mut rx_to_event_loop: Receiver<TransportCommand>,
+        config_path: Option<PathBuf>,
+    ) {
+        let runtime_wrapper = make_runtime_wrapper(vsomeip::runtime::get());
+
+        trace!("Entering command loop");
+        while let Some(cmd) = rx_to_event_loop.recv().await {
+            trace!("Received TransportCommand");
+
+            match cmd {
+                TransportCommand::RegisterListener(
```

Comment:
the handling of the commands should be done in separate functions

Reply:
They are. There's some light prep work and/or logging done before doing so.

See e.g. the `send_internal` function or others.

Reply:
Gonna call this one resolved, feel free to reopen if you'd like to discuss further.